src/main/java/com/epam/reportportal/extension/azure/AzureExtension.java

```diff
@@ -231,19 +233,40 @@ private void removeListeners() {
 
 	@Override
 	// Never called method. Connection is tested via the command.
-	public boolean testConnection(Integration system) {
+	public boolean testConnection(Integration integration) {
 		return false;
 	}
 
 	@Override
-	// TODO: Implement in the future
-	public Optional<Ticket> getTicket(String id, Integration system) {
-		return Optional.empty();
+	public Optional<Ticket> getTicket(String id, Integration integration) {
+		Map<String, Object> params = integration.getParams().getParams();
+		String organizationUrl = params.get("url").toString();
+		String projectName = params.get("project").toString();
+		String personalAccessToken = params.get("oauthAccessKey").toString();
```

Comment:
line 242-245 is a copypaste from other function. please make a function getParams which will give you a structured object. or introduce a class with parseFromParams function. the coolest way is to make generic parser for params, returning object of passed type=) but it's too much maybe

Reply:
I’m afraid there’s no way to avoid duplicate code. Because even if I return  a map or an object from a special `getParams()` method, I still need to retrieve fields from it (organizationUrl, projectName, personalAccessToken).

Reply:
![image](https://user-images.githubusercontent.com/12498983/129575198-2f6642f3-1208-4574-a08b-917a5022ad24.png)

Reply:
Introduced `getParams()` method as you suggested